### ISSUE TITLE
State is not being captured when using Snowflake destination

### ISSUE THREAD

**Issue by subodh1810** [airbytehq/airbyte#3811]
I set up a connection between MySQL CDC as source and Snowflake. The sync ran fine and all the records were fetched but the state was not saved. The logs had `2021-06-02 12:13:34 WARN (/tmp/workspace/46/0) DefaultReplicationWorker(run):182 - State capture: No state retained.`

This is only happening for Snowflake, when I ran the same connector against another destination like postgres and local csv, the state was captured and everything worked as expected.

As a result of this issue, there is no incremental sync happening, all the records are fetched in each sync

## Logs
[logs-46-0.txt](https://github.com/airbytehq/airbyte/files/6584431/logs-46-0.txt)
The above file is logs from sync, the logs from the docker did contain the state

```
    airbyte-scheduler   | 2021-06-02 12:13:06 INFO i.a.w.p.a.DefaultAirbyteStreamFactory(lambda$create$0):73 - {} - 2021-06-02 12:13:06 INFO i.a.i.s.m.MySqlSource(lambda$getIncrementalIterators$11):268 - {} - debezium state: {"mysql_db_history":"{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"ddl\":\"SET character_set_server=utf8mb4, collation_server=utf8mb4_0900_ai_ci;\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"DROP TABLE IF EXISTS `localhost_test`.`new_table`\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"DROP TABLE IF EXISTS `localhost_test`.`test_table`\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"DROP DATABASE IF EXISTS `localhost_test`\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"CREATE DATABASE `localhost_test` CHARSET utf8mb4 COLLATE utf8mb4_0900_ai_ci\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"USE `localhost_test`\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"CREATE TABLE `new_table` (\\n  `id` int NOT NULL,\\n  `val` varchar(200) DEFAULT NULL,\\n  PRIMARY KEY (`id`)\\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"CREATE TABLE `test_table` (\\n  `id` int NOT NULL,\\n  `val` varchar(200) DEFAULT NULL,\\n  `updated_at` timestamp(6) NULL DEFAULT NULL,\\n  PRIMARY KEY (`id`)\\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci\"}\n","mysql_cdc_offset":{"{\"schema\":null,\"payload\":[\"localhost_test\",{\"server\":\"localhost_test\"}]}":"{\"file\":\"binlog.000002\",\"pos\":4137}"}}
```    
## Steps to Reproduce
1. Set up a mysql in CDC as source
1. Set up snowflake as destination (loading method is AWS s3 Staging)
1. Run the sync and observe the logs

## Severity of the bug for you
Critical 

I think this is critical cause if it is happening for mysql, it might be happening for other connectors as well, I have not verified this

destination-snowflake:0.3.6
source-mysql:0.3.3
Airbyte version: dev

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200415526692745) by [Unito](https://www.unito.io)

**Comment by subodh1810** [comment]
Update, so I tested this against Snowflake destination with loading method as Standard Inserts and I did not experience this problem. 
[logs-47-0.txt](https://github.com/airbytehq/airbyte/files/6584543/logs-47-0.txt)
You can see in the logs that state was captured 
```
State capture: Updated state to: Optional[io.airbyte.config.State@6aa05513[state={"cdc":false,"cdc_state":{"state":{"mysql_db_history":"{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"ddl\":\"SET character_set_server=utf8mb4, collation_server=utf8mb4_0900_ai_ci;\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"DROP TABLE IF EXISTS `localhost_test`.`new_table`\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"DROP TABLE IF EXISTS `localhost_test`.`test_table`\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"DROP DATABASE IF EXISTS `localhost_test`\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"CREATE DATABASE `localhost_test` CHARSET utf8mb4 COLLATE utf8mb4_0900_ai_ci\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"USE `localhost_test`\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"CREATE TABLE `new_table` (\\n  `id` int NOT NULL,\\n  `val` varchar(200) DEFAULT NULL,\\n  PRIMARY KEY (`id`)\\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci\"}\n{\"source\":{\"server\":\"localhost_test\"},\"position\":{\"file\":\"binlog.000002\",\"pos\":4137,\"snapshot\":true},\"databaseName\":\"localhost_test\",\"ddl\":\"CREATE TABLE `test_table` (\\n  `id` int NOT NULL,\\n  `val` varchar(200) DEFAULT NULL,\\n  `updated_at` timestamp(6) NULL DEFAULT NULL,\\n  PRIMARY KEY (`id`)\\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci\"}\n","mysql_cdc_offset":{"{\"schema\":null,\"payload\":[\"localhost_test\",{\"server\":\"localhost_test\"}]}":"{\"file\":\"binlog.000002\",\"pos\":4137}"}}},"streams":[{"stream_name":"new_table","stream_namespace":"localhost_test","cursor_field":[]},{"stream_name":"test_table","stream_namespace":"localhost_test","cursor_field":[]}]}]]
```

**Comment by subodh1810** [comment]
Update: I built snowflake destination locally and used that image as a destination and ran the sync against it and everything worked fine. Turns out the issue started with this PR https://github.com/airbytehq/airbyte/pull/3290
In the PR we introduced checkpointing but the snowflake image was never published with the new logic https://github.com/airbytehq/airbyte/pull/3290#issuecomment-848304627
We didn't bump the snowflake docker image version in the PR and as a result the publish failed https://github.com/airbytehq/airbyte/runs/2669906026?check_suite_focus=true
error : 
```
image_name airbyte/destination-snowflake
airbyte/destination-snowflake:0.3.6 airbyte/destination-snowflake:0.3.6
latest_image airbyte/destination-snowflake:latest
You're trying to push a version that was already released (airbyte/destination-snowflake:0.3.6). Make sure you bump it up.
Error: Process completed with exit code 1.
```
